### PR TITLE
Avoid logging error monitoring k8s pods on restart

### PR DESCRIPTION
**Pull request by gp-jorge**
Fixes https://github.com/dotnet/orleans/issues/9044
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/orleans/pull/9048)

### CODE REVIEW

**Review by ReubenBond — APPROVED** [review]

File: src/Orleans.Hosting.Kubernetes/KubernetesClusterAgent.cs [hunk 1]
```diff
@@ -347,7 +347,7 @@ await foreach (var (eventType, pod) in pods.WatchAsync<V1PodList, V1Pod>(_shutdo
                         await Task.Delay(5000);
                     }
                 }
-                catch (Exception exception)
+                catch (Exception exception) when (!(_shutdownToken.IsCancellationRequested && (exception is TaskCanceledException || exception is OperationCanceledException)))
```
Comment (ReubenBond) [inline src/Orleans.Hosting.Kubernetes/KubernetesClusterAgent.cs:350]:
```suggestion
                catch (Exception exception) when (!(_shutdownToken.IsCancellationRequested && exception is OperationCanceledException))
```
`TaskCanceledException` derives from `OperationCanceledException` https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcanceledexception?view=net-8.0

**Comment by ReubenBond** [comment]
Thanks for the PR, @gp-jorge. Would you mind committing the suggestion above or enabling pushes to your branch?

**Comment by gp-jorge** [comment]
> Thanks for the PR, @gp-jorge. Would you mind committing the suggestion above or enabling pushes to your branch?

Thanks Reuben! Changed!

**Review by ReubenBond — APPROVED** [review]

**Comment by ReubenBond** [comment]
Fixes #8410

**Pull request merged**